### DISCUSSION TITLE
Disallowing only get or set method for property

### DISCUSSION THREAD

**Discussion by kkohno** [sebastienros/jint#1634] [Q&A]
Is there ways to disable access to get or set method of property only?

I need in strong sandboxing by interfaces and not any more. 
For example some interface can include only get method, but implementation can contains also set method.

```csharp
// the interface for sandbox
public interface IIndexedService
{
    int this[int index] { get; }
}

// the implementation
public sealed class IndexedService : IIndexedService
{
    List<int> _list = new List<int> { 100, 200, 300, 400, 500 };

    public int this[int index]
    {
        get => _list[index];
        set => _list[index] = value; // need to be ignored
    }
}
```

I use approach below, but it can ignore only whole property including get and set.
```csharp
options.SetTypeResolver(new TypeResolver {
    MemberFilter = resolver.FilterMember
});
```

**Comment by Zaitonn** [comment]
does making the setter of indexer `internal` or `private` or `protected` work in classes?

```cs
interface IExample
{
    int this[int index] { get; }
}

class Example : IExample
{
    public int this[int index]
    {
        get => 114514;
        internal set => Console.WriteLine(value);
    }
}
```

**Reply by kkohno** [reply]
yes, it works, but it can be a vulnerability if the implementation is not carefully done, or if there is already some library whose source codes are not available. The interface can also be implemented by a programmer from another department, and no one will look inside its code - the system must ensure security without delving into someone else's implementation
